engineyard/ey-cookbooks-stable-v6
# [FB-724] Env vars helper

#### Description of your patch
Move env vars helper to ey-lib/libraries/env_vars.rb
Check environment variables from all apps on fetch_env_var.
Use fetch_env_var on unattended-upgrades
Support EY_REDIS_ENABLED to enable redis using an environment variable
https://t3.fogbugz.com/f/cases/724/Move-EnvVars-Helper

#### Recommended Release Notes
Set EY_REDIS_ENABLED to install redis on a utility instance.

#### Estimated risk
Medium

#### Components involved
Environment variables
Redis
Unattended upgrades

#### Description of testing done
See QA instructions

#### QA Instructions
1. Redis
Add EY_REDIS_ENABLED environment variable. Use any value. 
Boot a utility instance named redis. 
Check that redis is running (using redis-cli or systemctl status redis-server.

2. Unattended upgrades
Add EY_ENABLE_UNATTENDED_UPGRADES environment variable. Set it to true. 
Check /etc/apt/apt.conf.d/20auto-upgrades. It should have APT::Periodic::Unattended-Upgrade "1";
